### PR TITLE
chore(test-data): create `Channels` presets

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

This PR creates test data presets for Channels, based on First Contact's requirements surrounding sample project data.

## Description

These models were simple and straightforward enough to configure. I opted for snapshot testing, as these are expected to be generated deterministically.

Additional comments inline.

### CODE REVIEW

File: models/channel/src/channel-draft/presets/hub.spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { TChannelDraft, TChannelDraftGraphql } from '../../types';
+import hub from './hub';
```
Comment (jaikamat) [inline models/channel/src/channel-draft/presets/hub.spec.ts:2]: How do we feel about this naming? I went off each entity's `key`

File: models/channel/src/channel-draft/presets/hub.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  it('should match graphql snapshot', () => {
+    const channel = hub().buildGraphql<TChannelDraftGraphql>();
+
+    expect(channel).toMatchInlineSnapshot(`
+      {
+        "__typename": "ChannelDraft",
+        "address": undefined,
+        "custom": undefined,
```
Comment (jaikamat) [inline models/channel/src/channel-draft/presets/hub.spec.ts:41]: I wonder if we'll need some sort of prefiltering to clear these `undefined` values when we make a query. Right now we sort of generate _everything_.

File: models/channel/src/channel-draft/presets/hub.spec.ts (hunk 1 above)
Comment (jaikamat) [reply models/channel/src/channel-draft/presets/hub.spec.ts:2]: Great strategy!

Comment (jaikamat) [reply models/channel/src/channel-draft/presets/hub.spec.ts]: @tylermorrisford see 1283eb4c39d7ae7291fc293dfcf25d1c6722c49c 😄

File: models/channel/src/channel-draft/presets/sample-data-fashion/store-other.spec.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import storeOther from './store-other';
+
+describe('store-other channel', () => {
+  it('should match REST snapshot', () => {
+    const channel = storeOther().build<TChannelDraft>();
+
+    expect(channel).toMatchInlineSnapshot(`
+      {
+        "address": undefined,
+        "custom": undefined,
```
Comment (jaikamat) [inline models/channel/src/channel-draft/presets/sample-data-fashion/store-other.spec.ts:11]: I wonder if we'll need some sort of prefiltering to clear these undefined fields when we make a query. Right now we sort of generate everything.

Comment (stephsprinkle) [reply models/channel/src/channel-draft/presets/sample-data-fashion/store-other.spec.ts:11]: After checking out other preset tests, they don't use `toMatchInlineSnapshot`, which means you don't have to bother checking fields that aren't set.

File: models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { roles } from '../../../constants';
+import ChannelDraft from '../../builder';
+
+const storeOther = () =>
+  ChannelDraft()
+    .key('store_other')
+    .address(undefined)
+    .geoLocation(undefined)
+    .custom(undefined)
```
Comment (stephsprinkle) [inline models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts:10]: Not needed

File: models/channel/src/channel-draft/presets/sample-data-fashion/store-usa.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { roles } from '../../../constants';
+import ChannelDraft from '../../builder';
+
+const storeUsa = () =>
+  ChannelDraft()
+    .key('store_usa')
+    .address(undefined)
+    .geoLocation(undefined)
+    .custom(undefined)
```
Comment (stephsprinkle) [inline models/channel/src/channel-draft/presets/sample-data-fashion/store-usa.ts:10]: Not needed

File: models/channel/src/channel-draft/presets/sample-data-fashion/hub.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { roles } from '../../../constants';
+import ChannelDraft from '../../builder';
+
+const hub = () =>
+  ChannelDraft()
+    .key('hub')
+    .address(undefined)
+    .geoLocation(undefined)
+    .custom(undefined)
```
Comment (stephsprinkle) [inline models/channel/src/channel-draft/presets/sample-data-fashion/hub.ts:10]: Not needed

File: models/channel/src/channel-draft/presets/index.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
+import * as SampleDataFashion from './sample-data-fashion';
 import withInventorySupplyAndProductDistributionRoles from './with-inventory-supply-and-product-distribution-roles';
 import withInventorySupplyAndProductDistributionRolesNoAddress from './with-inventory-supply-and-product-distribution-roles-no-address';
 import withInventorySupplyRole from './with-inventory-supply-role';
 import withInventorySupplyRoleNoAddress from './with-inventory-supply-role-no-address';
 import withProductDistributionRole from './with-product-distribution-role';
 
 const presets = {
```
Comment (stephsprinkle) [inline models/channel/src/channel-draft/presets/index.ts:8]: Do we need to update the `channel/index.ts` file to include exports of the draft presets? Without it, what would the import look like?

File: models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts (hunk 4 above)
Comment (jaikamat) [reply models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts:10]: I might have been thinking too much about implementation here, is the plan to use these in the resolvers like this:
```
import presets from '../presets'

const storeUsa = presets.SampleDataFashion.storeUsa;

// Ignore this obviously wrong syntax, it's just to illustrate a point
makeRequestToCoCo('url', {
    variables: {
        draft: {
            name: storeUsa.name,
            description: storeUsa.description,
            key: storeUsa.key,
            roles: storeUsa.roles,
        }
    }
})```

or like this

```
import presets from '../presets'

makeRequestToCoCo('url', {
    variables: {
        draft: { ...presets.SampleDataFashion.storeUsa() }
    }
})
```

I was thinking of the latter, hence my obsession with ensuring the presets were of a specific shape - but that might be too obsessive 😬

Comment (stephsprinkle) [reply models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts:10]: Definitely the latter! Those three undefined fields aren't needed because they aren't required fields and we aren't using them.

Comment (jaikamat) [reply models/channel/src/channel-draft/presets/sample-data-fashion/store-other.ts:10]: Ohhhh. I'm picking up what you're putting down. We just won't include these fields in the query, so they will be ignored.

File: models/channel/src/channel-draft/presets/index.ts (hunk 7 above)
Comment (jaikamat) [reply models/channel/src/channel-draft/presets/index.ts:8]: Addressed in 0850dc85538824e43a82848df7469440000c7a33

File: models/channel/src/channel-draft/presets/index.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import * as SampleDataFashion from './sample-data-fashion';
```
Comment (stephsprinkle) [inline models/channel/src/channel-draft/presets/index.ts]: Why did you remove the `* as`?

Comment (jaikamat) [reply models/channel/src/channel-draft/presets/index.ts]: When requiring in from other packages, `* as` had the following effect:

```
import {draftPresets, presets} from '@commercetools-test-data/channel';

presets.clothesStore

draftPresets.SampleDataFashion.default.storeOther // with `* as`
draftPresets.SampleDataFashion.storeOther // without `* as`
```

`channel-draft.presets.index.ts` already has a `default` export, so the `* as` syntax is superfluous. [These docs ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#namespace_import)helped me understand what was going on here, too

Comment (stephsprinkle) [reply models/channel/src/channel-draft/presets/index.ts]: Ah! The zone presets need to be updated then because they use the `* as` syntax.

**Review by stephsprinkle — APPROVED** [review]

File: models/channel/src/channel-draft/presets/index.ts (hunk 8 above)
Comment (jaikamat) [reply models/channel/src/channel-draft/presets/index.ts]: Done! Addressed in 1b506a6e9f118d2e6614cc066b244fc44c9e1d2b

**Review by tdeekens — APPROVED** [review]

File: models/channel/src/channel-draft/presets/index.ts [hunk 9]
```diff
@@ -1,10 +1,12 @@
+import SampleDataFashion from './sample-data-fashion';
 import withInventorySupplyAndProductDistributionRoles from './with-inventory-supply-and-product-distribution-roles';
 import withInventorySupplyAndProductDistributionRolesNoAddress from './with-inventory-supply-and-product-distribution-roles-no-address';
 import withInventorySupplyRole from './with-inventory-supply-role';
 import withInventorySupplyRoleNoAddress from './with-inventory-supply-role-no-address';
 import withProductDistributionRole from './with-product-distribution-role';
 
 const presets = {
+  SampleDataFashion,
```
Comment (emmenko) [inline models/channel/src/channel-draft/presets/index.ts:9]: Nit: lowercase for consistency? `sampleDataFashion`

**Review by emmenko — APPROVED** [review]
Nice!

File: models/zone/src/zone-draft/presets/index.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import * as SampleDataFashion from './sample-data-fashion';
+import SampleDataFashion from './sample-data-fashion';
```
Comment (tylermorrisford) [inline models/zone/src/zone-draft/presets/index.ts:2]: 🙏

**Review by tylermorrisford — APPROVED** [review]

File: models/channel/src/channel-draft/presets/index.ts (hunk 9 above)
Comment (jaikamat) [reply models/channel/src/channel-draft/presets/index.ts]: [Sounds good!](https://github.com/commercetools/test-data/pull/205/commits/cd126a4c140b9ae7a53d78fb295079ec14cb73d1)
